tests/suites/test_suite_psa_crypto_slot_management.data

```diff
@@ -13,54 +13,15 @@ transient_slot_lifecycle:0:0:PSA_KEY_TYPE_RAW_DATA:"0123456789abcdef0123456789ab
 Transient slot, check after restart with live handles
 transient_slot_lifecycle:0:0:PSA_KEY_TYPE_RAW_DATA:"0123456789abcdef0123456789abcdef":CLOSE_BY_SHUTDOWN
 
```

Comment:
I've restored all the test cases. We still support open/close thus this clean-up was indeed premature. I've also restored some code in `persistent_slot_lifecycle`. The code is eventually not the same as in development (usage of key identifier instead of key handle essentially) but the checks are the same but when checking that an handle is not valid anymore after a close because this is not true anymore.

Reply:
Addressed in commits:
Revert "tests: Remove some invalid persistent key handle tests"
tests: slot mgmt: Restore close related code